localizationkit/tests/placeholder_token_explanation.py

```diff
@@ -0,0 +1,95 @@
+"""Placeholder token explanation."""
+
+from typing import Any, Dict, List, Set
+
+from localizationkit.tests.test_case import LocalizationTestCase
+
+
+class PlaceholderTokenExplanation(LocalizationTestCase):
+    """Check the placeholder tokens in strings have explanation in comments."""
+
+    @classmethod
+    def name(cls) -> str:
+        return "placeholder_token_explanation"
+
+    @classmethod
+    def default_settings(cls) -> Dict[str, Any]:
+        return {"always": False}
+
+    def run_test(self) -> List[str]:
+
+        violations = []
+
+        for string in self.collection.localized_strings:
+            tokens = string.tokens()
```

Comment:
We have a choice to make here:

1) We assume that tokens passed in are unique and rely on other tests to ensure that. i.e. Something like "Hello %@, it is %@ today" wouldn't be passed in (or it would, but the results for this test are invalid). 

2) We call into common functionality to ensure that this is the case, and return the violations if not. 

I think option 1 is the easiest and cleanest. We just need to be sure that it is documented in the README. 

If that's the case, this can all be changed to something like:

```python
tokens = set(string.tokens())
comment_tokens = set(string.comment_tokens())

extra_in_tokens = tokens - comment_tokens
extra_in_comments = comment_tokens - tokens

if len(extra_in_tokens) != 0:
    violations.append(f"Tokens are not described in the comment ({extra_in_tokens}): {string}")

if len(extra_in_comments) != 0:
    violations.append(f"Extra tokens appear in the comment ({extra_in_comments}): {string}")
```

Reply:
I agree with ^ The initial thought I had aligns with this where we keep it simple and down to just mismatch between number of placeholder tokens and their explanation in comments. Then I looked over other tests and rules and thought we should make it more stricter. I agree that brings in duplication of the checks, we can start with what you suggested and then move to making the rule for `token_position_identifiers` more flexible that takes in an arg to validate string and/or its comment for tokens